chromium_src/components/page_load_metrics/page_load_metrics_embedder_base.cc

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/page_load_metrics/browser/page_load_metrics_embedder_base.h"
+
+#include "base/timer/timer.h"
+
+#include "components/page_load_metrics/browser/observers/core_page_load_metrics_observer.h"
+#include "components/page_load_metrics/browser/observers/use_counter_page_load_metrics_observer.h"
+#include "components/page_load_metrics/browser/page_load_tracker.h"
+
+#include "brave/components/brave_perf_predictor/browser/buildflags/buildflags.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_PERF_PREDICTOR)
+#include "brave/components/brave_perf_predictor/browser/perf_predictor_page_metrics_observer.h"
+using brave_perf_predictor::PerfPredictorPageMetricsObserver;
+#endif
+
+namespace page_load_metrics {
+
+PageLoadMetricsEmbedderBase::PageLoadMetricsEmbedderBase(
```

Comment:
we never want to copy entire files. I think probably what makes sense here is to override `InitializePageLoadMetricsForWebContents` in `PageLoadMetricsEmbedder` and return a subclass of `PageLoadMetricsEmbedder` that overrides `RegisterEmbedderObservers` to add `PerfPredictorPageMetricsObserver` which also deals with the dep issue for `components/brave_perf_predictor` in `components/page_load_metrics`